### DISCUSSION TITLE
SQL injection

### DISCUSSION THREAD

**Discussion by jossmanbr** [pocketbase/pocketbase#3034] [Q&A]
There are precautionary measures that should be taken regarding SQL injection

**Comment by ganigeorgiev** [comment]
Please elaborate, I'm not sure that I understand what exactly is the question here.

In general when building manually queries you should check "Binding parameters":

- (Go) https://pocketbase.io/docs/go-database/#binding-parameters
- (JS) https://pocketbase.io/docs/js-database/#binding-parameters

**Reply by calebfaruki** [reply]
Very helpful docs. But this begs the question: if you feed a raw template literal into the filter parameter of the JS SDK, does it sanitize as well? Or only when you use the bind parameters?

**Reply by ganigeorgiev** [reply]
The above links are for when interacting directly with the database (aka. when using Go or the JS `pb_hooks`).

The JS SDK is used for interacting with the [Web APIs](https://pocketbase.io/docs/api-records/) and the PocketBase server always treat the submitted request data as untrusted and applies extra validation and/or normalization.

Therefore when using the JS SDK on the client-side in terms of security it doesn't matter whether you'll escape the request parameters or not. But note that when using the JS SDK in a node/deno/bun/etc. **server-side** context, depending on your use case this may not be the case and may have to sanitize untrusted user input that you pass to the JS SDK calls to prevent string injection attacks. To simplify the "sanitization" the JS SDK comes with a helper `pb.filter` method. For more details please refer to https://github.com/pocketbase/js-sdk#binding-filter-parameters.
